addons/mail/static/tests/new/thread/thread_tests.js

```diff
@@ -181,3 +184,74 @@ QUnit.test(
         assert.containsNone(target, ".o-mail-thread-date-separator");
     }
 );
+
+QUnit.test(
+    "Scroll position is kept when navigating from one channel to another",
+    async function (assert) {
+        const pyEnv = await startServer();
+        const firstChannelId = pyEnv["mail.channel"].create({ name: "channel-1" });
+        const secondChannelId = pyEnv["mail.channel"].create({ name: "channel-2" });
+        // Fill both channels with random messages in order for the scrollbar to
+        // appear.
+        pyEnv["mail.message"].create(
+            Array(40)
+                .fill(0)
+                .map((_, idx) => ({
+                    body: "Non Empty Body ".repeat(25),
+                    message_type: "comment",
+                    model: "mail.channel",
+                    res_id: idx & 1 ? firstChannelId : secondChannelId,
+                }))
+        );
+        const { click, openDiscuss } = await start({
+            discuss: {
+                context: {
+                    active_id: `mail.channel_${firstChannelId}`,
+                },
+            },
+        });
+        await openDiscuss();
+        let currentThread = document.querySelector(".o-mail-thread");
+        const firstThreadScrollTop = currentThread.scrollHeight / 2;
+        currentThread.scrollTo({ top: firstThreadScrollTop, left: 0 });
+        await click(`.o-mail-category-item[data-channel-id=${secondChannelId}]`);
+        currentThread = document.querySelector(".o-mail-thread");
+        const secondThreadScrollTop = currentThread.scrollHeight / 3;
+        currentThread.scrollTo({ top: secondThreadScrollTop, left: 0 });
+        await click(`.o-mail-category-item[data-channel-id=${firstChannelId}]`);
+        currentThread = document.querySelector(".o-mail-thread");
+        assert.ok(isScrolledTo(currentThread, firstThreadScrollTop));
+        await click(`.o-mail-category-item[data-channel-id=${secondChannelId}]`);
+        currentThread = document.querySelector(".o-mail-thread");
+        assert.ok(isScrolledTo(currentThread, secondThreadScrollTop));
```

Comment:
This `currentThread` variable makes the test much harder to read. Better not Introduce it.